airflow/www/views.py

```diff
@@ -785,13 +787,25 @@ def index(self):
             # Remove the reset_tags=reset from the URL
             return redirect(url_for("Airflow.index"))
 
+        if arg_lastrun_filter == "reset_filter":
+            flask_session[FILTER_LASTRUN_COOKIE] = None
+            return redirect(url_for("Airflow.index"))
+
         cookie_val = flask_session.get(FILTER_TAGS_COOKIE)
         if arg_tags_filter:
             flask_session[FILTER_TAGS_COOKIE] = ",".join(arg_tags_filter)
         elif cookie_val:
             # If tags exist in cookie, but not URL, add them to the URL
             return redirect(url_for("Airflow.index", tags=cookie_val.split(",")))
 
+        cookie_val = flask_session.get(FILTER_LASTRUN_COOKIE)
```

Comment:
Overwriting `cookie_val` seems risky. IMO all `cookie_val`s should be updated to be explicit, for example `filter_tags_cookie_value` and `filter_lastrun_cookie_value`.